### PR TITLE
Handle token approvals and deposit tx

### PR DESCRIPTION
**Pull request by luizakp**
https://www.loom.com/share/f2706ec58e234b238b66d665c2f6e6c1

### CODE REVIEW

File: packages/next-app/src/hooks/useTransaction.ts [hunk 1]
```diff
@@ -90,6 +92,11 @@ const NOTIFICATION_MAP = {
 };
 
 export const NOTIFICATION_MAP_INTERNAL_BALANCES = {
+  [TransactionStatus.AUTHORIZING]: {
+    title: "Approve this transaction",
+    description: "Waiting for your wallet approvement",
```
Comment (ribeirojose) [inline packages/next-app/src/hooks/useTransaction.ts:97]:
```suggestion
    description: "Waiting for your approval",
```
same below

File: packages/next-app/src/hooks/useTransaction.ts [hunk 2]
```diff
@@ -100,6 +107,11 @@ export const NOTIFICATION_MAP_INTERNAL_BALANCES = {
     description: "Your transaction is being made",
     variant: NotificationVariant.NOTIFICATION,
   },
+  [TransactionStatus.CONFIRMING]: {
+    title: "Great!",
+    description: "The approval was a success!",
```
Comment (ribeirojose) [inline packages/next-app/src/hooks/useTransaction.ts:112]:
```suggestion
    description: "The approval was successful!",
```
the copies sound a bit off to me -- I can help out with that later on but would take a read at how other apps are doing it.

File: packages/schema/index.ts [hunk 3]
```diff
@@ -81,11 +84,20 @@ export const getInternalBalanceSchema = ({
           message: "Token amount must be greater than 0",
         });
       }
-      if (Number(val) > totalBalance) {
-        ctx.addIssue({
-          code: z.ZodIssueCode.custom,
-          message: "Amount exceeds total balance",
-        });
+      if (typeof totalBalance === "string") {
```
Comment (ribeirojose) [inline packages/schema/index.ts:87]: 👀

File: packages/next-app/src/hooks/useTransaction.ts [hunk 4]
```diff
@@ -272,28 +317,37 @@ export function useInternalBalancesTransaction({
...
-        chainId: chain!.id,
-        txHash: hash,
-      });
-      setNotification(
-        NOTIFICATION_MAP_INTERNAL_BALANCES[TransactionStatus.SUBMITTING]
-      );
-      setTransactionUrl(txUrl);
-    }
-    handleTransactionStatus();
+    handleTransactionStatus({ hash });
```
Comment (ribeirojose) [inline packages/next-app/src/hooks/useTransaction.ts:350]: 👌 nice refactor

File: packages/next-app/src/hooks/useTransaction.ts [hunk 5]
```diff
@@ -260,8 +274,39 @@ export function useInternalBalancesTransaction({
...
+      abi: erc20ABI,
+      functionName: "approve",
+      args: [
+        vaultAddress[5],
+        parseFixed(
+          submitData?.tokenAmount ? submitData.tokenAmount : "0",
+          tokenDecimals
+        ),
+      ],
+    });
```
Comment (ribeirojose) [inline packages/next-app/src/hooks/useTransaction.ts:295]: 👁️ could we generate a wagmi hook for this? We'd get warned by the usepreparedtransactionwrite if it'd fail

**Review by ribeirojose — APPROVED** [review]
🙇

File: packages/next-app/src/hooks/useTransaction.ts (hunk 2 above)
Comment (luizakp) [reply packages/next-app/src/hooks/useTransaction.ts:112]: Thanks! =)

**Pull request merged**